torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -333,15 +333,8 @@ class ProcessGroupNCCL : public ProcessGroup {
       return fut;
     }
 
-    // Checks cudaEventQuery with cudaEvents. Returns true if a FutureError was
-    // recorded or the entire operation is completed on the GPU.
     bool completed() const override {
-      if (error_) {
-        return true;
-      }
-      // Checking the work's corresponding CUDA events' status
-      auto ret = cudaEventQuery((*cudaEvents_)[0]);
-      return ret != cudaErrorNotReady || ret == cudaSuccess;
+      return true;
```

Comment:
there is no regression here, and I agree that user code won't trigger any problem. But C++ code can still use the second ctor of `FutureNCCL` to create an empty future without marking it as completed?

Reply:
Yes, correct. As I mentioned earlier, the second constructor is "meant" to be private and only invoked by `then()`. If you want we can make that explicit. However, in #48505 we'll fix this issue for good by having FutureNCCL share the same logic as ivalue::Future (and thus have a real complete-vs-incomplete status).